pkg/controller/mcpserver_controller.go

```diff
@@ -545,6 +547,39 @@ func (r *MCPReconciler) discoverServersFromHTTPRoutes(
 	return &serverInfo, nil
 }
 
+// buildServerInfoForHostnameBackend handles Istio Hostname backendRef for external services
+func (r *MCPReconciler) buildServerInfoForHostnameBackend(
+	httpRoute *gatewayv1.HTTPRoute,
+	mcpServer *mcpv1alpha1.MCPServer,
+	backendRef gatewayv1.HTTPBackendRef,
+	namespace string,
+	httpRouteName string,
+) (*ServerInfo, error) {
+	if len(httpRoute.Spec.Hostnames) == 0 {
+		return nil, fmt.Errorf("HTTPRoute %s/%s must have at least one hostname", namespace, httpRouteName)
+	}
+
+	externalHost := string(backendRef.Name)
+	port := "443"
+	if backendRef.Port != nil {
+		port = fmt.Sprintf("%d", *backendRef.Port)
+	}
+
+	path := mcpServer.Spec.Path
+	endpoint := fmt.Sprintf("https://%s%s", net.JoinHostPort(externalHost, port), path)
```

Comment:
I am worried about safety here.  Can we add a check that externalHost is a valid DNS name?  A malicious user might create a backendRef.Name such as `badcountry.com:443/path/`.  (There might be other checks, but adding enforcement here will make it easier to audit this code.)